libcaf_core/test/config_value.cpp

```diff
@@ -81,13 +86,739 @@ config_value cfg_lst(Ts&&... xs) {
+    WHEN("using get_as with integer types") {
+      THEN("CAF fails to convert the real to an integer") {
+        CHECK_EQ(get_as<uint64_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<uint32_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<int32_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<uint16_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<int16_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<uint8_t>(x), sec::conversion_failed);
+        CHECK_EQ(get_as<int8_t>(x), sec::conversion_failed);
+      }
+    }
+  }
+  GIVEN("config_values of null, URI, boolean, list or dictionary") {
+    WHEN("using get_as with integer types") {
+      THEN("conversion fails") {
+        CHECK_EQ(get_as<int64_t>(cv_null), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_true), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_false), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_uri), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_list), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_dict), sec::conversion_failed);
+      }
+    }
+  }
+}
+
+SCENARIO("get_as can convert config values to floating point numbers") {
+  GIVEN("a config value x with value 1.79769e+308") {
+    auto x = config_value{1.79769e+308};
+    WHEN("using get_as with floating point types") {
+      THEN("conversion fails if bounds checks fail") {
+        CHECK_EQ(get_as<long double>(x), 1.79769e+308);
+        CHECK_EQ(get_as<double>(x), 1.79769e+308);
+        CHECK_EQ(get_as<float>(x), sec::conversion_failed);
+      }
+    }
+  }
+  GIVEN("a config value x with value \"3e7\"") {
+    auto x = config_value{"3e7"s};
+    WHEN("using get_as with floating point types") {
+      THEN("CAF parses the string and converts the value") {
+        CHECK_EQ(get_as<long double>(x), 3e7);
+        CHECK_EQ(get_as<double>(x), 3e7);
+        CHECK_EQ(get_as<float>(x), 3e7f);
+      }
+    }
+  }
+  GIVEN("a config value x with value 123") {
+    auto x = config_value{123};
+    WHEN("using get_as with floating point types") {
+      THEN("CAF converts the value") {
+        CHECK_EQ(get_as<long double>(x), 123.0);
+        CHECK_EQ(get_as<double>(x), 123.0);
+        CHECK_EQ(get_as<float>(x), 123.f);
+      }
+    }
+  }
+  GIVEN("config_values of null, URI, boolean, list or dictionary") {
+    WHEN("using get_as with floating point types") {
+      THEN("conversion fails") {
+        CHECK_EQ(get_as<int64_t>(cv_null), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_true), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_false), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_uri), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_list), sec::conversion_failed);
+        CHECK_EQ(get_as<int64_t>(cv_empty_dict), sec::conversion_failed);
```

Comment:
Not a floating point type.

Reply:
Correct. 🧐

Reply:
I think the we check twice now that the conversion to `int64_t` fails. I should have been more clear in my first comment on that, sorry. The first block that checks this is in lines 253 to 265. The comment was probably correct, but the conversion type should be different?

Reply:
Yeah, looks like it. Thanks for the extra context, this makes it clear now.